client/src/components/UnsavedChangesIndicator/UnsavedChangesIndicatorTimer.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import UnsavedChangesIndicator from './UnsavedChangesIndicator';
+
+/**
+ * A container component for UnsavedChangesIndicator that tracks the time since
+ * it was marked dirty, and updates the indicator level accordingly.
+ * By default it has a 5 minute notice level and 10 minute warning level.
+ */
+const UnsavedChangesIndicatorTimer = ({
+  isDirty = false,
+  minutes = {
+    notice: 5,
+    warning: 10,
+  },
+  UnsavedChangesIndicatorComponent = UnsavedChangesIndicator,
+}) => {
+  const [level, setLevel] = useState('none');
+  const [startTime, setStartTime] = useState(null);
+
+  /**
+   * Reset timer when isDirty changes
+   */
+  useEffect(() => {
+    if (isDirty) {
+      setStartTime(Date.now());
+      setLevel('none');
+    } else {
+      setStartTime(null);
+      setLevel('none');
+    }
+  }, [isDirty]);
+
+  /**
+   * Check elapsed time and update level accordingly
+   */
+  useEffect(() => {
+    if (!isDirty || startTime === null) {
+      return () => {};
+    }
+    const checkElapsedTime = () => {
+      const elapsed = Date.now() - startTime;
+      const warningMs = minutes.warning ? minutes.warning * 60000 : null;
+      const noticeMs = minutes.notice ? minutes.notice * 60000 : null;
```

Comment:
If the `minutes` values are both falsy we should just set level to "none" and not set an interval.